montage.py

```diff
@@ -121,13 +121,10 @@ def exportframe(self):
         self.frames = kinematic.makeframes()
         if not self.frames:
             return
-        firstpixindex = self.frames.keys()[0]
-
-        x, y, width, height = self.mfdraw.get_allocation()
-        pixmap = gtk.gdk.Pixmap(self.mfdraw.window, width, height)
-        self._draw_frame(firstpixindex, pixmap)
-        pixbuf = gtk.gdk.Pixbuf(gtk.gdk.COLORSPACE_RGB, False, 8, width, height)
-        gtk.gdk.Pixbuf.get_from_drawable(pixbuf, pixmap, pixmap.get_colormap(), 0, 0, 0, 0, width, height)
+        firstpixindex = list(self.frames.keys())[0]
+        window = Gdk.get_default_root_window()
+        x,y, width , height = window.get_geometry()
+        pixbuf = Gdk.pixbuf_get_from_window(window, x, y, width, height)
```

Comment:
You've ported from GTK 2 to GTK 3 and reimplemented a screenshot feature.  Activity.get_preview does this now; can it be used instead?

Reply:
Thanks I will try with get_preview, hope that will work.

Reply:
@quozl I wonder why we need to give screenshots feature in this activity , and also a button in toolbar for the same. We have already option in toolkit to take screenshot with keyboard keys. Why it's necessary?

Reply:
Does it offer more than the Sugar feature?  Was the feature added to the activity before it was added to Sugar?  The feature in Sugar is whole display including toolbar.  Does this feature include toolbar?

Reply:
I am not sure about displaying toolbar as I couldn't test it. However the way I implemented here, it includes toolbar.

Reply:
Sugar had the global key screenshot feature added in 2009.  See https://github.com/sugarlabs/sugar/blame/master/extensions/globalkey/screenshot.py

Sugar had the frame display icon screenshot feature added six years ago.  See https://github.com/sugarlabs/sugar/blame/master/extensions/deviceicon/display.py#L180

get_preview was mentioned because the code in this activity is similar to get_preview.  I'm not surprised you didn't find when get_preview was added.  Toolkit has a second repository, for the GTK 2 source.  In that, https://github.com/sugarlabs/sugar-toolkit/blame/master/src/sugar/activity/activity.py#L663 shows get_preview was made available (by public API) in 2009.  The activity would have been written for the previous version of Sugar at the time.  The education department of the country that focused on this activity delayed upgrading Sugar on their laptops.

So to answer my questions;

> Does it offer more than the Sugar feature?

Yes, it takes a snapshot of the animation window, not the canvas, not the display.

> Was the feature added to the activity before it was added to Sugar?

Yes.

> The feature in Sugar is whole display including toolbar. Does this feature include toolbar?

No.

Reply:
Thanks @quozl . Maybe 1dee462 is a good fix for this activity. I have tested it looks like this , it is opened in ImageViewer.

![Screenshot from 2020-08-07 11-57-01](https://user-images.githubusercontent.com/45252461/89616304-cb493f00-d8a5-11ea-9aeb-878c2997fec6.png)

Reply:
The grey box top right is surprising.  Is that the toolbar button palette?

Reply:
Yes it is.

Reply:
>The grey box top right is surprising. Is that the toolbar button palette?

It can be seen only with smaller screens. Because whole screen becomes too congested.
I tried to solve it in ec83aec. It looks fine no tool tip can be seen now.